### PR TITLE
LPC55S69: Add PWM support

### PR DESCRIPTION
**Pull request by mmahadevan108**

### CODE REVIEW

File: soc/arm/nxp_lpc/lpc55xxx/soc.h [hunk 1]
```diff
@@ -27,10 +27,11 @@
 #define IOCON_PIO_DIGITAL_EN 0x0100u  /*!<@brief Enables digital function */
 #define IOCON_PIO_FUNC0 0x00u         /*!<@brief Selects pin function 0 */
 #define IOCON_PIO_FUNC1 0x01u         /*!<@brief Selects pin function 1 */
-#define IOCON_PIO_FUNC2 0x02u         /*!<@brief Selects pin function 1 */
+#define IOCON_PIO_FUNC2 0x02u         /*!<@brief Selects pin function 2 */
+#define IOCON_PIO_FUNC4 0x04u         /*!<@brief Selects pin function 4 */
 #define IOCON_PIO_FUNC5 0x05u         /*!<@brief Selects pin function 5 */
```
Comment (dleach02) [inline soc/arm/nxp_lpc/lpc55xxx/soc.h:32]: is this file something we pull from the SDK? I'm curious as to why we needed to update this file (for future reference)?

**Review by dleach02 — APPROVED** [review]

File: soc/arm/nxp_lpc/lpc55xxx/soc.h (hunk 1 above)
Comment (mmahadevan108) [reply soc/arm/nxp_lpc/lpc55xxx/soc.h:32]: These defines are from the SDK.

**Review by danieldegrasse — APPROVED** [review]

**Comment by dleach02** [comment]
@mmahadevan108, can you resolve the conflicts here?

**Review by henrikbrixandersen — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by mmahadevan108** [comment]
Thank you, done

**Pull request merged**